aws/data_source_aws_kinesis_stream_consumer_test.go

```diff
@@ -0,0 +1,35 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccAWSKinesisStreamConsumerDataSource_basic(t *testing.T) {
+	var stream kinesis.StreamDescription
+	var consumer kinesis.ConsumerDescription
+	config := createAccKinesisStreamConsumerConfig()
```

Comment:
interesting usage! just for maintainability, and current recommendations, we should however prefer to reference functions that are more explicit when defining resources and/or data-sources. I'll update these tests with updated examples.